<launch>
    <!-- roll pid controller -->
    <node name="controller" pkg="pid" type="controller" ns="pid_roll" >
        <param name="Kp" value="-3.0" />
        <param name="Ki" value="-0.0" /> 
        <param name="Kd" value="-1.8" />
        <param name="upper_limit" value="0.6" />
        <param name="lower_limit" value="-0.6" />
        <param name="windup_limit" value="1.0" />
        <param name="max_loop_frequency" value="20.0" />
        <param name="min_loop_frequency" value="20.0" />
        <remap from="setPoint" to="/zero_setpoint" />
    </node>

    <!-- pitch pid controller -->
    <node name="controller" pkg="pid" type="controller" ns="pid_pitch">
        <param name="Kp" value="-3.0" />
        <param name="Ki" value="-0.0" />
        <param name="Kd" value="-1.8" />
        <param name="upper_limit" value="0.6" />
        <param name="lower_limit" value="-0.6" />
        <param name="windup_limit" value="0" />
        <param name="max_loop_frequency" value="20.0" />
        <param name="min_loop_frequency" value="20.0" />
        <remap from="setpoint" to="/zero_setpoint" />
    </node>

    <!-- thrust pid controller -->
    <node name="controller" pkg="pid" type="controller" ns="pid_thrust">
        <param name="Kp" value="-3.0" />
        <param name="Ki" value="0" />
        <param name="Kd" value="-1.8" />
        <param name="upper_limit" value="0.0" /> 
        <param name="lower_limit" value="0.0" />
        <param name="windup_limit" value="0.4" />
        <param name="max_loop_frequency" value="20.0" />
        <param name="min_loop_frequency" value="20.0" />
        <remap from="setpoint" to="/zero_setpoint" />
    </node>

    <!-- yaw pid controller -->
    <node name="controller" pkg="pid" type="controller" ns="pid_yaw">
        <param name="Kp" value="1.3" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.8" />
        <param name="upper_limit" value="0.0" />
        <param name="lower_limit" value="-0.0" />
        <param name="windup_limit" value="0.0" />
        <param name="max_loop_frequency" value="20.0" />
        <param name="min_loop_frequency" value="20.0" />
        <param name="angle_error" value="true" />
        <remap from="setpoint" to="/zero_setpoint" />
    </node>

</launch>